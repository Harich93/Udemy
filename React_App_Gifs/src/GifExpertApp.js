import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ( { defaulCategoria = [] }) => {

    const [categorias, setCategoria] = useState( defaulCategoria );

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setCategoria={ setCategoria } />
            <hr />
            
            <ol>
                {
                    categorias.map( cats => (
                        <GifGrid 
                            key={ cats }
                            categoria={ cats }
                        />
                    ) )
                }
            </ol>
        </>
    )
}

export default GifExpertApp
