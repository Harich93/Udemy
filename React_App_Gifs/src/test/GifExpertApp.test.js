import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    
    test('Snapshot', () => {
        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot
    });
    
    test('Debe mostrar una lista de categorias', () => {
        
        const categorias = [ 'Goku', 'Days Gone'];
        const wrapper = shallow( <GifExpertApp defaulCategoria={ categorias } /> );

        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categorias.length );

    });
    
})
