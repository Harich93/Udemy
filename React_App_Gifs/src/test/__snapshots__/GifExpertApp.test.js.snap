// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategoria={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Goku"
      key="Goku"
    />
    <GifGrid
      categoria="Days Gone"
      key="Days Gone"
    />
  </ol>
</Fragment>
`;
